### PR TITLE
reworked built-in glob configs to configure from json snippets

### PR DESCRIPTION
**Pull request by seancpeters**
Regarding performance differences from before and after this change, I did some nominal testing and didn't see any real differences. I ran the mvc1.1 template 4 times against the code before and after my changes, with negligible timing differences:

New way:
 Content generation time 1158.7476
 Content generation time 976.8154
 Content generation time 1069.5988
 Content generation time 1022.9127
   
Old way
Content generation time 957.0962
Content generation time 1107.2132
Content generation time 975.9659
Content generation time 1142.1048

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs [hunk 1]
```diff
@@ -32,130 +42,40 @@ public IEnumerable<IOperationProvider> ConfigureFromJObject(JObject rawConfigura
             {
                 operations = ConditionalBlockCommentConfig.ConfigureFromJObject(rawConfiguration);
             }
-            else
+            else if (string.Equals(commentStyle, "builtin", StringComparison.OrdinalIgnoreCase))
```
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs]: Add a check to catch loops in definitions - a built-in could theoretically reference another built-in, or itself - which is an authoring error.

**Comment by seancpeters** [comment]
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test Ubuntu x64 Release Build please
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by mlorbetske** [comment]
We should take a look at what, if any impact there is to perf for this. Also, I assume this is the first step in moving to producing the core configuration from the JSON directly or is the intention to only build it in pieces?

**Comment by seancpeters** [comment]
Regarding perf, see my initial PR comment - it's not definitive in any way. I can try to isolate the timings for the glob setup and compare.

Yes, this is just a first step toward more json-based configuration, but I'm still on the fence about the final form of core config setup. Using json files is more "pure", so that's probably where we'll land.

**Comment by seancpeters** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please
@dotnet-bot test Debian8.2 x64 Debug Build please
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by seancpeters** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please

**Pull request closed without merging**